Xamarin.Forms.Platform.Android/Material/MaterialProgressBarRenderer.cs

```diff
@@ -142,51 +146,11 @@ void UpdateColors()
 			if (Element == null || Control == null)
 				return;
 
-			Color progressColor = Element.ProgressColor;
-			Color backgroundColor = Element.BackgroundColor;
-
-			var defaultProgress = MaterialColors.Light.PrimaryColor;
-
-			if (progressColor.IsDefault && backgroundColor.IsDefault)
-			{
-				// reset everything to defaults
-				ProgressTintList = ColorStateList.ValueOf(defaultProgress);
-				ProgressBackgroundTintList = ColorStateList.ValueOf(defaultProgress);
-				ProgressBackgroundTintMode = PorterDuff.Mode.SrcIn;
-			}
-			else if (progressColor.IsDefault && !backgroundColor.IsDefault)
-			{
-				// handle the case where only the background is set
-				var background = backgroundColor.ToAndroid();
-
-				// TODO: Potentially override primary color to match material design.
-				ProgressTintList = ColorStateList.ValueOf(defaultProgress);
-				ProgressBackgroundTintList = ColorStateList.ValueOf(background);
-
-				// TODO: Potentially override background alpha to match material design.
-				ProgressBackgroundTintMode = PorterDuff.Mode.SrcOver;
-			}
-			else if (!progressColor.IsDefault && backgroundColor.IsDefault)
-			{
-				// handle the case where only the progress is set
-				var progress = progressColor.ToAndroid();
-
-				ProgressTintList = ColorStateList.ValueOf(progress);
-				ProgressBackgroundTintList = ColorStateList.ValueOf(progress);
-				ProgressBackgroundTintMode = PorterDuff.Mode.SrcIn;
-			}
-			else
-			{
-				// handle the case where both are set
-				var background = backgroundColor.ToAndroid();
-				var progress = progressColor.ToAndroid();
-
-				ProgressTintList = ColorStateList.ValueOf(progress);
-				ProgressBackgroundTintList = ColorStateList.ValueOf(background);
-
-				// TODO: Potentially override alpha to match material design.
-				ProgressBackgroundTintMode = PorterDuff.Mode.SrcOver;
-			}
+			var primary = Element.ProgressColor.IsDefault ? _defaultPrimaryColor : Element.ProgressColor.ToAndroid();
+			ProgressTintList = ColorStateList.ValueOf(primary);
+			var background = Element.BackgroundColor.IsDefault ? _defaultBackgroundColor : Element.BackgroundColor.ToAndroid();
+			ProgressBackgroundTintList = ColorStateList.ValueOf(background);
+			ProgressBackgroundTintMode = PorterDuff.Mode.SrcIn;
```

Comment:
I think we need to have a fallback to follow material and if only the progress color is set, then we have the lighter background of that color - unless specifically set. Right now, it looks that if I set the color to purple, only the main color is set, the track remains black - immediately breaking the material design. If we want to allow the user to break the guides, then we need to let them do the work to get there.